scripts/apps/monitoring/directives/MonitoringView.js

```diff
@@ -14,19 +14,41 @@ export function MonitoringView($rootScope, authoringWorkspace, pageTitle, $timeo
             state: '='
         },
         link: function(scope, elem) {
-            var containerElem = elem.find('.content-list');
+            const containerElem = elem.find('.content-list');
 
             pageTitle.setUrl(_.capitalize(gettext(scope.type)));
 
             scope.shouldRefresh = true;
-            scope.view = 'compact';
+            scope.view = 'compact'; // default view
 
             scope.workspaces = workspaces;
             scope.$watch('workspaces.active', (workspace) => {
                 scope.workspace = workspace;
             });
             workspaces.getActive();
 
+            scope.desks = desks;
+            scope.$watch('desks.desks', () => {
```

Comment:
it should probably watch `scope.$watch(desks.getCurrentDesk.bind(desks), (currentDesk) => {`